### DISCUSSION TITLE
Iterate cache entities

### DISCUSSION THREAD

**Discussion by bottleofwater** [urql-graphql/urql#1843] [Q&A] [answered]
Is it possible in Graphcache to iterate over the cache entities inside an `update` spec? I want `createBook` to invalidate all the `books` fields from all the `Collection` entities that are currently cached, but I don't know how to retrieve the collections' entity keys to use with `cache.invalidate`.

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

I think you're looking for [inspectFields](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#inspecting-arbitary-entities)

**Reply by bottleofwater** [reply]
Hey! From my understanding `inspectFields` expects an entity key too, so it needs an ID to work with. In my case I don't have the ID of the collections currently in cache. Am I mistaken? I tried doing this, but it returns an empty array:

```js
cache.inspectField('Collection')
```

Only `inspectField('Query')` returns something.

**Reply by JoviDeCroock** [reply]
That's correct yes, we can't look for partial entities currently

**Reply by bottleofwater** [reply]
Should I open an feature request issue?
